iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/convert.inl

```diff
@@ -72,41 +74,59 @@ inline bool convert::fromString<std::string>(const char* v, std::string& dest) n
 template <>
 inline bool convert::fromString<char>(const char* v, char& dest) noexcept
 {
-    if (strlen(v) != 1u)
+    if (strlen(v) != 1U)
     {
         std::cerr << v << " is not a char" << std::endl;
         return false;
     }
+
+    /// @NOLINTJUSTIFICATION encapsulated in abstraction
+    /// @NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
     dest = v[0];
     return true;
 }
 
+/// @NOLINTJUSTIFICATION iox-#1196 todo convert must be refactored
+/// @NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers)
 template <>
 inline bool convert::fromString<string<100>>(const char* v, string<100>& dest) noexcept
 {
     dest = string<100>(TruncateToCapacity, v);
     return true;
 }
+/// @NOLINTEND(cppcoreguidelines-avoid-magic-numbers)
 
 inline bool convert::stringIsNumber(const char* v, const NumberType type) noexcept
 {
+    /// @NOLINTJUSTIFICATION encapsulated in abstraction
+    /// @NOLINTBEGIN(cppcoreguidelines-pro-bounds-pointer-arithmetic)
     if (v[0] == '\0')
+    {
         return false;
+    }
 
     bool hasDot = false;
 
-    for (uint32_t i = 0u; v[i] != '\0'; ++i)
+    for (uint32_t i = 0U; v[i] != '\0'; ++i)
     {
-        if (v[i] >= 48 && v[i] <= 57) // 48 == ascii 0, 57 == ascii 9
+        constexpr char ASCII_INTEGER_ZERO = 48;
+        constexpr char ASCII_INTEGER_NINE = 57;
+        constexpr char ASCII_SYMBOL_PLUS = 43;
+        constexpr char ASCII_SYMBOL_MINUS = 45;
+        constexpr char ASCII_SYMBOL_DOT = 46;
```

Comment:
Why not just `'0'`, `'9'`, `'+'`, etc.? The compiler knows it anyway but it is easier to grasp for carbon based mortals ;)